packages/xstate-react/src/createActorContext.tsx

```diff
@@ -1,38 +1,80 @@
 import * as React from 'react';
-import { RestParams, useInterpret } from './useInterpret';
+import { useInterpret } from './useInterpret';
 import { useActor as useActorUnbound } from './useActor';
 import { useSelector as useSelectorUnbound } from './useSelector';
-import { ActorRefFrom, AnyStateMachine, EmittedFrom } from 'xstate';
+import {
+  ActorRefFrom,
+  AnyStateMachine,
+  AreAllImplementationsAssumedToBeProvided,
+  EmittedFrom,
+  InternalMachineOptions,
+  InterpreterOptions,
+  Observer,
+  StateFrom
+} from 'xstate';
+import { UseMachineOptions } from './useMachine';
 
 export function createActorContext<TMachine extends AnyStateMachine>(
   machine: TMachine,
-  ...[options = {}, observerOrListener]: RestParams<TMachine>
+  interpreterOptions?: InterpreterOptions,
+  observerOrListener?:
+    | Observer<StateFrom<TMachine>>
+    | ((value: StateFrom<TMachine>) => void)
 ): {
   useActor: () => ReturnType<typeof useActorUnbound<ActorRefFrom<TMachine>>>;
   useSelector: <T>(
     selector: (snapshot: EmittedFrom<TMachine>) => T,
     compare?: (a: T, b: T) => boolean
   ) => T;
   useActorRef: () => ActorRefFrom<TMachine>;
-  Provider: (props: {
-    children: React.ReactNode;
-    machine?: TMachine | (() => TMachine);
-  }) => React.ReactElement<any, any>;
+  Provider: (
+    props: {
+      children: React.ReactNode;
+      machine?: TMachine | (() => TMachine);
+    } & (AreAllImplementationsAssumedToBeProvided<
```

Comment:
Why can't we have the simpler type here? At least without `AreAllImplementationsAssumedToBeProvided`

Reply:
The added type-oriented tests wouldn't pass. It's quite nice that we validate if the user has configured all implementations or not (the cost for that is the readability of those types though 😬)